zerver/lib/addressee.py

```diff
@@ -21,9 +22,26 @@ def user_profiles_from_unvalidated_emails(emails: Iterable[str], realm: Realm) -
         user_profiles.append(user_profile)
     return user_profiles
 
-def get_user_profiles(emails: Iterable[str], realm: Realm) -> List[UserProfile]:
+def user_profiles_from_ids(ids: Iterable[Any]) -> List[UserProfile]:
+    user_profiles = []
+    for user_id in ids:
+        try:
+            user_profile = get_user_profile_by_id(user_id)
```

Comment:
This should probably be `get_user_profile_by_id_in_realm` or similar; otherwise you can message people in other Zulip orgs with this!  We should include a test that it we don't have this bug, since it's a classic thing that could go wrong with this sort of API, so even if we do it right at first, we want to protect against a problematic refactor.